guardian/workflow-frontend
# Only call for content once at startup

## What does this change?

Thanks to [this PR](https://github.com/guardian/workflow-frontend/pull/281), a request for content won't render if another request has occured since the initial request went out. The issue is that, when starting up Workflow, 3 requests for `content` get made. These are:

1. An intentional call for content from the `startPolling` function
2. The instantiation of the `selectedDesk` listener, which invokes the code within the callback
3. The instantiation of the `selectedView` listener, which does the same as above

Since each call to `content` returns large payloads (around 11mb on PROD), this results in Workflow waiting for the third call to respond before rendering anything.

 This PR ensures that the watchers only get invoked if there is a difference between the old and new values, which there isn't when the watcher gets invoked. This should offer significant performance improvements to Workflow at startup. 

## How to test

Spin up Workflow / go to CODE, open the Network tab and look for calls to `content` at startup. You should see there's only one, rather than the three before.

## How can we measure success?
Loading articles in Workflow at startup takes 1/3rd of the time that it currently takes.

## Images

BEFORE:
![image](https://user-images.githubusercontent.com/25747336/91185961-5455d800-e6e6-11ea-8cd3-21d8ffa10ae7.png)

AFTER:
![image](https://user-images.githubusercontent.com/25747336/91185914-4acc7000-e6e6-11ea-9716-598e57fe05f5.png)
